cmd/up.go

```diff
@@ -103,9 +103,10 @@ func ClusterUp(ctx context.Context, dialersOptions hosts.DialersOptions, flags c
 		return rebuildClusterWithRotatedCertificates(ctx, dialersOptions, flags, svcOptionsData)
 	}
 	// if we need to rotate the encryption key, do so and then return
+	// note that we rotate the encryption key only when updating an existing cluster that has secret encryption enabled
+	// all other cases will be handled later by reconciling the encryption provider config
 	if kubeCluster.RancherKubernetesEngineConfig.RotateEncryptionKey {
-		// rotate the encryption key only when updating an existing cluster
-		if clusterState.CurrentState.RancherKubernetesEngineConfig != nil {
+		if clusterState.CurrentState.RancherKubernetesEngineConfig != nil && clusterState.CurrentState.RancherKubernetesEngineConfig.Services.KubeAPI.SecretsEncryptionConfig.Enabled {
```

Comment:
`clusterState.CurrentState.RancherKubernetesEngineConfig.Services.KubeAPI.SecretsEncryptionConfig` is a pointer. Should we have a `nil`-check on that too?

Reply:
Good catch! 
Fixed.

Reply:
I will squash commits into one after getting your approval.